src/main/java/org/cryptomator/linux/quickaccess/NautilusBookmarks.java

```diff
@@ -0,0 +1,93 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+@Priority(100)
+@CheckAvailability
+public class NautilusBookmarks implements QuickAccessService {
+
+	private static final int MAX_FILE_SIZE = 4096;
+	private static final Path BOOKMARKS_FILE = Path.of(System.getProperty("user.home"), ".config/gtk-3.0/bookmarks");
+	private static final Path TMP_FILE = BOOKMARKS_FILE.resolveSibling("bookmarks.cryptomator.tmp");
+	private static final Lock BOOKMARKS_LOCK = new ReentrantReadWriteLock().writeLock();
+
+	@Override
+	public QuickAccessService.QuickAccessEntry add(Path target, String displayName) throws QuickAccessServiceException {
+		String entryLine = "file://" + target.toAbsolutePath() + " " + displayName;
+		try {
+			BOOKMARKS_LOCK.lock();
+			if (Files.size(BOOKMARKS_FILE) > MAX_FILE_SIZE) {
+				throw new IOException("File %s exceeds size of %d bytes".formatted(BOOKMARKS_FILE, MAX_FILE_SIZE));
+			}
+			//by reading all lines, we ensure that each line is terminated with EOL
+			var entries = Files.readAllLines(BOOKMARKS_FILE, StandardCharsets.UTF_8);
+			entries.add(entryLine);
+			Files.write(TMP_FILE, entries, StandardCharsets.UTF_8, StandardOpenOption.WRITE, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
+			Files.move(TMP_FILE, BOOKMARKS_FILE, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE);
+			return new NautilusQuickAccessEntry(entryLine);
+		} catch (IOException e) {
+			throw new QuickAccessServiceException("Adding entry to Nautilus bookmarks file failed.", e);
+		} finally {
+			BOOKMARKS_LOCK.unlock();
+		}
+	}
+
+	static class NautilusQuickAccessEntry implements QuickAccessEntry {
+
+		private final String line;
+		private volatile boolean isRemoved = false;
+
+		NautilusQuickAccessEntry(String line) {
+			this.line = line;
+		}
+
+		@Override
+		public void remove() throws QuickAccessServiceException {
+			try {
+				BOOKMARKS_LOCK.lock();
+				if (isRemoved) {
+					return;
+				}
+				if (Files.size(BOOKMARKS_FILE) > MAX_FILE_SIZE) {
+					throw new IOException("File %s exceeds size of %d bytes".formatted(BOOKMARKS_FILE, MAX_FILE_SIZE));
+				}
+				var entries = Files.readAllLines(BOOKMARKS_FILE);
+				if (entries.remove(line)) {
+					Files.write(TMP_FILE, entries, StandardCharsets.UTF_8, StandardOpenOption.WRITE, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
+					Files.move(TMP_FILE, BOOKMARKS_FILE, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE);
+				}
+				isRemoved = true;
+			} catch (IOException e) {
+				throw new QuickAccessServiceException("Removing entry from Nautilus bookmarks file failed", e);
+			} finally {
+				BOOKMARKS_LOCK.unlock();
+			}
+		}
+	}
+
+	@CheckAvailability
+	public static boolean isSupported() {
+		try {
+			var nautilusExistsProc = new ProcessBuilder().command("test", "`command -v nautilus`").start();
```

Comment:
@SailReal Linux question: Can there be multiple file explorers installed in parallel? Then this command would assume nautilus to be present (which isn't even incorrect) despite it not being used. ATM not a problem, but if there were multiple different implementations the highest priority would win.

Reply:
Way ahead of you!

> Can there be multiple file explorers installed in parallel?

yes, but the interface allows to list all supported implementations, see [integrations-api](https://github.com/cryptomator/integrations-api/blob/4bfb945a08267d6b2038363ad6024690f9a91d9a/src/main/java/org/cryptomator/integrations/quickaccess/QuickAccessService.java#L52-L54). Hence, the service consumer can pick one.

Reply:
Dammn, guess I need to extend the quick access SPI with a `getDisplayName()` method 😅

Reply:
Since the use of a display name to let the user pick alternatives from the UI is a recurring pattern, we might want to add an annotation for the display name? This would allow to include these services in a list without fully instantiating them.

Reply:
> without fully instantiating them

This can conflict with non-static `@CheckAvailability` methods.

But it is indeed a reoccuring pattern and will persist, since on Linux we always have possible alternatives (not necessarily implemented, but still though).

Reply:
@overheadhunter i drafted an interface in https://github.com/cryptomator/integrations-api/pull/39 for the purpose of a displayName.

Reply:
> This can conflict with non-static `@CheckAvailability` methods.

No, there is a [clear order in which available implementations are filtered](https://github.com/cryptomator/integrations-api/blob/4bfb945a08267d6b2038363ad6024690f9a91d9a/src/main/java/org/cryptomator/integrations/common/IntegrationsLoader.java#L48-L52), but each check is independent of the other.

Reply:
Display name added in 7040c0eeaa364e82f9d2d3f3065de8350ee003e5